src/main/java/com/example/java/epam/brayan/services/CreateTicketService.java

```diff
@@ -3,34 +3,43 @@
 import com.example.java.epam.brayan.data.entities.Event;
 import com.example.java.epam.brayan.data.entities.Ticket;
 import com.example.java.epam.brayan.data.entities.User;
-import com.example.java.epam.brayan.data.repositories.EventRepository;
 import com.example.java.epam.brayan.data.repositories.TicketRepository;
-import com.example.java.epam.brayan.data.repositories.UserRepository;
 import com.example.java.epam.brayan.services.data.NewTicket;
 import com.example.java.epam.brayan.services.mappers.NewTicketMapper;
-import jakarta.persistence.EntityNotFoundException;
+import com.example.java.epam.brayan.services.validators.CreateTicketValidator;
 import jakarta.transaction.Transactional;
 import lombok.RequiredArgsConstructor;
+import org.springframework.data.util.Pair;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class CreateTicketService {
-    private final UserRepository userRepository;
-    private final EventRepository eventRepository;
+    private final CreateTicketValidator createTicketValidator;
     private final TicketRepository ticketRepository;
+    private final DeductUserAccountBalanceService deductUserAccountBalanceService;
     private final NewTicketMapper newTicketMapper;
 
     @Transactional
     public Ticket createTicket(NewTicket newTicket) {
-        Ticket ticket = newTicketMapper.toTicket(newTicket);
-        User user = userRepository.findById(newTicket.getUserId())
-                .orElseThrow(() -> new EntityNotFoundException("User not found"));
-        Event event = eventRepository.findById(newTicket.getEventId())
-                .orElseThrow(() -> new EntityNotFoundException("Event not found"));
+        Pair<User, Event> validatedUserAndEvent = createTicketValidator.validateNewTicket(newTicket);
```

Comment:
Looks like a violation of Command Query and SRP pattern.
The function is about validation and refers the Ticket but it magically returns some objects and additionally to expected ticket contains Event.

What I would recommend is to move up the validation on higher level.

The service should consider the valid input.

Take a look at Spring validation framework. For many cases it could be sufficient just put some annotations on data class fields and annotate the input param in the controller with `@Valid` annotation. 
For more complicated cases or WebFlux however you should consider some separate service to do the validation.